common/lib/opentelemetry/common/http/request_attributes.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Common
+    module HTTP
+      # RequestAttributes contains instrumentation helpers for http instrumentation client requests
+      module RequestAttributes
+        QUERY_PARAM_START_KEY = '?'
+        def from_request(method:, config: {}, uri: nil, scheme: nil, target: nil, url: nil, hostname: nil, port: nil)
```

Comment:
Can we also add `host: nil` here? The semantic conventions say this should contain the value of the HTTP host header. While there are other things we could add, the `http.host` attribute is one of the attributes that

> **MUST** be provided **at span creation time** (when provided at all), so they can be considered for sampling decisions

It would be great if we could add `ip: nil` as well. IDK if that's available in any of the HTTP client gems, though.